src/Build/BuildCheck/Infrastructure/BuildCheckConnectorLogger.cs

```diff
@@ -26,55 +26,58 @@ public void Initialize(IEventSource eventSource)
 
     private void EventSource_AnyEventRaised(object sender, BuildEventArgs e)
     {
-        if (e is ProjectEvaluationFinishedEventArgs projectEvaluationFinishedEventArgs)
+        if (!buildCheckManager.isRestore)
```

Comment:
I'd suggest to invert the condition and return - it'll reduce nesting and as well simplify the diff.

```suggestion
        if (buildCheckManager.isRestore)
        {
           return;
        }
```